### PR TITLE
feat(core): changes for coyo app integration

### PR DESCRIPTION
**Pull request by manyx**

### CODE REVIEW

File: angular/projects/catalyst/src/lib/catalyst.module.ts [hunk 1]
```diff
@@ -17,7 +21,7 @@ const CatComponents = [
 export class CatalystModule {
 
   static forRoot(): ModuleWithProviders<CatalystModule> {
-    defineCustomElements();
+    defineCustomElements(window);
```
Comment (vardidzel) [inline angular/projects/catalyst/src/lib/catalyst.module.ts:24]: I'm just wondering if can `defineCustomElements` be called without `window`?

File: angular/projects/catalyst/package.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
   },
   "peerDependencies": {
     "@angular/core": ">=12.0.0",
-    "rxjs": ">=6.6.0"
+    "rxjs": ">=6.6.0",
+    "@haiilo/catalyst": ">=0.0.1"
```
Comment (fynnfeldpausch) [inline angular/projects/catalyst/package.json:15]: Thinking about it, I guess it is not a peerDependency, as the core and angular wrapper a closely coupled. So it would rather be a real dependency with a fixed version. Upside: No need to specify the dependency on COYO side anymore.

File: angular/projects/catalyst/src/lib/catalyst.module.ts [hunk 3]
```diff
@@ -5,7 +5,11 @@ import * as Components from './directives/proxies';
 const CatComponents = [
   Components.CatButton,
   Components.CatIcon,
-  Components.CatSpinner
+  Components.CatSpinner,
```
Comment (fynnfeldpausch) [inline angular/projects/catalyst/src/lib/catalyst.module.ts:8]: Was this autogenerated?

File: angular/projects/catalyst/src/lib/catalyst.module.ts (hunk 1 above)
Comment (fynnfeldpausch) [reply angular/projects/catalyst/src/lib/catalyst.module.ts:24]: Same question. The examples I saw were calling the function without window?

File: angular/tsconfig.json [hunk 4]
```diff
@@ -5,8 +5,8 @@
     "baseUrl": "./",
     "paths": {
       "catalyst": [
-        "dist/catalyst/catalyst",
-        "dist/catalyst"
+        "dist/catalyst",
+        "dist/catalyst/*"
```
Comment (fynnfeldpausch) [inline angular/tsconfig.json:9]: Was this autogenerated?

File: core/src/components/cat-button/cat-button.scss [hunk 5]
```diff
@@ -11,8 +11,10 @@
 .cat-button {
   font: inherit;
   display: flex;
+  flex: 1 1 auto;
```
Comment (fynnfeldpausch) [inline core/src/components/cat-button/cat-button.scss:14]: please revert this change. we will fix it separately

File: core/src/components/cat-skeleton/readme.md [hunk 6]
```diff
@@ -15,6 +15,15 @@
 | `variant` | `variant` | The rendering style of the skeleton.                                                                                                                   | `"body" \| "circle" \| "head" \| "rectangle" \| "square"` | `'rectangle'` |
 
 
+## CSS Custom Properties
```
Comment (fynnfeldpausch) [inline core/src/components/cat-skeleton/readme.md:18]: auto-generated?

**Review by fynnfeldpausch — CHANGES_REQUESTED** [review]

File: angular/projects/catalyst/src/lib/catalyst.module.ts (hunk 1 above)
Comment (manyx) [reply angular/projects/catalyst/src/lib/catalyst.module.ts:24]: I guess it can work without it, I saw some examples that used window so I added it but I'll remove it.

File: angular/projects/catalyst/src/lib/catalyst.module.ts (hunk 3 above)
Comment (manyx) [reply angular/projects/catalyst/src/lib/catalyst.module.ts:8]: No, I added them to include the new components that we added.

File: core/src/components/cat-button/cat-button.scss (hunk 5 above)
Comment (manyx) [reply core/src/components/cat-button/cat-button.scss:14]: done

File: core/src/components/cat-skeleton/readme.md (hunk 6 above)
Comment (manyx) [reply core/src/components/cat-skeleton/readme.md:18]: yes

File: angular/projects/catalyst/package.json (hunk 2 above)
Comment (manyx) [reply angular/projects/catalyst/package.json]: moved it to dependencies.

File: angular/projects/catalyst/src/lib/catalyst.module.ts (hunk 3 above)
Comment (fynnfeldpausch) [reply angular/projects/catalyst/src/lib/catalyst.module.ts:8]: ah, makes sense. i thought the stencil CLI already takes care of it but apparently it does not :D
